src/test/java/com/github/tomakehurst/wiremock/SnapshotAcceptanceTest.java

```diff
@@ -0,0 +1,274 @@
+import static com.github.tomakehurst.wiremock.testsupport.TestHttpHeader.withHeader;
+import static com.github.tomakehurst.wiremock.testsupport.WireMatchers.equalToJson;
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertThat;
+
+public class SnapshotAcceptanceTest extends AcceptanceTestBase {
+    private WireMockServer proxyingService;
+    private WireMockTestClient proxyingTestClient;
+
+    public void init() {
+        proxyingService = new WireMockServer(wireMockConfig()
+            .dynamicPort()
+            .withRootDirectory("src/test/resources/empty"));
+        proxyingService.start();
+        proxyingService.stubFor(proxyAllTo("http://localhost:" + wireMockServer.port()));
+
+        proxyingTestClient = new WireMockTestClient(proxyingService.port());
+        wireMockServer.stubFor(any(anyUrl()).willReturn(ok()));
+    }
+
+    @After
+    public void proxyServerShutdown() {
+        proxyingService.stop();
+    }
+
+    private static final String DEFAULT_SNAPSHOT_RESPONSE =
+            "[                                                           \n" +
+            "    {                                                       \n" +
+            "        \"request\" : {                                     \n" +
+            "            \"url\" : \"/foo/bar/baz\",                     \n" +
+            "            \"method\" : \"GET\"                            \n" +
+            "        },                                                  \n" +
+            "        \"response\" : {                                    \n" +
+            "            \"status\" : 200                                \n" +
+            "        }                                                   \n" +
+            "    },                                                      \n" +
+            "    {                                                       \n" +
+            "        \"request\" : {                                     \n" +
+            "            \"url\" : \"/foo/bar\",                         \n" +
+            "            \"method\" : \"GET\"                            \n" +
+            "        },                                                  \n" +
+            "        \"response\" : {                                    \n" +
+            "            \"status\" : 200                                \n" +
+            "        }                                                   \n" +
+            "    }                                                       \n" +
+            " ]                                                            ";
+
+    @Test
+    public void returnsRequestsWithDefaultOptions() throws Exception {
+        proxyingTestClient.get("/foo/bar", withHeader("A", "B"));
+        proxyingTestClient.get("/foo/bar/baz", withHeader("A", "B"));
+
+        assertThat(
+            proxyingTestClient.snapshot("{ \"persist\": false }"),
```

Comment:
It will persist the stub mappings to disk, since it persists by default. If there were an acceptance test for that, then there'd have to be logic to delete the saved mappings after the test in order to guarantee test isolation. I noticed that `AcceptanceTestBase` has a `setupServerWithTempFileRoot()` method for that, but I can't use it here because I want to apply it to the `proxyingService` instance, not the backend WiremockServer. Also, it doesn't seem to ensure the files are deleted at the end of each test. Seems to me the easiest fix would be to change `setupServerWithTempFileRoot()` to just return the temporary file root, so it'd be renamed `createTempFileRoot()`, and have a `@After` hook to delete it.

`SnapshotTaskTest` has some a unit test for persistence (`persistsStubMappingWhenPersistSet()`), so at least that's covered.